include/deal.II/base/symmetric_tensor.h

```diff
@@ -421,7 +421,7 @@ namespace internal
     {
       using tensor_type = const ::dealii::SymmetricTensor<rank, dim, Number>;
 
-      using reference = Number;
+      using reference = const Number &;
```

Comment:
I guess the question is whether we want to expose the underlying memory address (as with a reference) or not (then, the right type would be `const Number;`). I agree with the current patch.